docs/csharp/language-reference/builtin-types/struct.md

```diff
@@ -0,0 +1,56 @@
+---
+title: "Structure types - C# reference"
+ms.date: 02/20/2020
+f1_keywords: 
+  - "struct_CSharpKeyword"
+helpviewer_keywords: 
+  - "struct keyword [C#]"
+  - "structs [C#], struct keyword"
+ms.assetid: ff3dd9b7-dc93-4720-8855-ef5558f65c7c
+---
+# Structure types (C# reference)
+
+A *structure type* (or *struct type*) is a [value type](value-types.md) that encapsulates data and related functionality. As the following example shows, you use the `struct` keyword to define a structure type:
+
+[!code-csharp[struct example](~/samples/csharp/language-reference/builtin-types/StructType.cs#StructExample)]
+
+Because a structure type is a value type, a variable of a structure type contains an instance of the type. That is called value semantics. By default, variables are copied on assignment, passing an argument to a method, or returning a method result. In the case of a structure-type variable, an instance of the type is copied (for the examples, see [Value types](value-types.md)). In high-performance scenarios that involve large structure types, you can avoid value copying and pass around an alias to a value-type variable. For information about how to do that, see [Write safe and efficient C# code](../../write-safe-efficient-code.md).
+
+Typically, you define a structure type when you need to represent a small data structure that has value semantics. For example, .NET uses structure types to represent a number (both [integer](integral-numeric-types.md) and [real](floating-point-numeric-types.md)), a [Boolean value](bool.md), a [Unicode character](char.md), a [time instance](xref:System.DateTime). If you need a type with reference semantics (that is, a variable of the type contains a reference to an instance of that type, not the instance itself), define a [class](../keywords/class.md).
+
+## Limitations with the design of a structure type
+
+When you design a structure type, you have the same capabilities as with a [class](../keywords/class.md) type, with the following exceptions:
+
+- You cannot declare a parameterless constructor. Every structure type already provides an implicit parameterless constructor that returns the [default value](default-values.md) of the type.
+
+- You cannot initialize an instance field at its declaration. However, you can initialize a [static](../keywords/static.md) or [const](../keywords/const.md) field at its declaration.
+
+- A structure type cannot inherit from other class or structure type and it cannot be the base of a class. However, a structure type can implement [interfaces](../keywords/interface.md).
+
+- You cannot declare a [finalizer](../../programming-guide/classes-and-structs/destructors.md) within a structure type.
+
+## Instantiation of a structure type
```

Comment:
> ...using a combination of any accessible constructor and accessible property initializers

That reads to me as that I can call a constructor and then use properties to initialize remaining members. For example, for given struct:
```csharp
    public struct Coords
    {
        public double X { get; set; }
        public double Y;

        public Coords(double y) => Y = y;

        public override string ToString() => $"{X:F2} {Y:F2}";
    }
```

I would be able to call 
```csharp
Coords a = new Coords(y: 5);
a.X = 4;
```

However, the above struct declaration doesn't compile because `X` is not assigned in the constructor. So one must use only `new` with properly defined constructor that initializes all instance data members. 

Probably, I misunderstand you @BillWagner. What do you mean by "using a combination of any accessible constructor and accessible property initializers"?

Reply:
Ahh. I was thinking something like this.

From the struct author perspective, all fields must be initialized in all declared constructors:

```csharp
public struct Coords
{
    public double X { get; set; }
    public double Y;

    private Coords(double x, double y) => (X, Y) = (x, y);
    public Coords(double y) : this(0.0, y) { }
}
```

And then from the perspective of callers, they can use constructor args or accessible field accessors to set values:

```csharp
Coords a = new Coords(y: 5) {X = 4 };
```

Looking at this conversation, it's probably wise to separate those two perspectives.

Great conversation and discussion. I am confident this will lead to a great final version. 👯‍♂

Reply:
Great! Thank you for the clarification and the example!